app/services/spree/feeds/xml.rb

```diff
@@ -4,6 +4,74 @@
 module Spree
   module Feeds
     class XML < Spree::Feeds::Base
+
+      def doc_header
+        doc_parts.first
+      end
+
+      def doc_footer
+        doc_parts.last
+      end
+
+      def doc_parts
+        doc_skeleton.split('<item_content/>')
+      end
+
+      # The XML document with no items, we then split this into
+      # header and footer
+      def doc_skeleton
+        @doc_skeleton ||= Ox::Builder.new(indent: 2) { |b|
+          b.instruct(:xml, version: '1.0', encoding: 'UTF-8')
+          b.element('rss', version: '2.0', 'xmlns:g' => 'http://base.google.com/ns/1.0') do
+            b.element('channel') do
+              b.element('title') { b.text store.name }
+              b.element('link') { b.text store.url }
+              b.element('description') { b.text "Find out about new products on http://#{store.url} first!" }
+              b.element('item_content')
+            end
+          end
+        }
+      end
+
+      # Supply an Enumerator so that this can be evaluated in a controller
+      # action to stream the XML
+      def item_elements
+        Enumerator.new do |y|
+        end
```

Comment:
should this be an inline block since it's short and empty?

Reply:
Oh, that's a dead method that needs to be removed. Thanks for the pointer.